user_tools/src/spark_rapids_pytools/pricing/price_provider.py

```diff
@@ -32,38 +32,52 @@ class PriceProvider:
     An abstract class that represents interface to retrieve costs of hardware configurations.
     """
     region: str
-    pricing_config: JSONPropertiesContainer
-    cache_file: str = field(default=None, init=False)
-    resource_url: str = field(default=None, init=False)
+    # pricing_config: JSONPropertiesContainer
+    pricing_configs: dict #[str, JSONPropertiesContainer]
+    # cache_file: str = field(default=None, init=False)
+    cache_files: dict = field(default=None, init=False) #[str, str]
```

Comment:
you can initialize a dictionary in dataclasses like this 
```
    my_dict: dict = field(default_factory=dict, init=False)
```
This will create an empty dictionary by default that you can later update by inserting keys.

Reply:
Changes updated. Thank you!